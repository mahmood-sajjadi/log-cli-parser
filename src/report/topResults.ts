export function getTopResults<T>(arr: T[], len: number): T[] {
  const l = Math.min(len, arr.length);
  return arr.slice(0, l);
}
