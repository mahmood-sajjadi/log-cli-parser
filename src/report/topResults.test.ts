import { getTopResults } from './topResults';

describe('getTopResults', () => {
  test('WHEN list BIGGER than request PROVIDE what requested', () => {
    const list = ['abc', 'mno', 'xyz'];
    const result = getTopResults(list, 2);
    expect(result).toHaveLength(2);
    expect(result).toMatchSnapshot();
  });

  test('WHEN list SMALLER than request PROVIDE entire array', () => {
    const list = ['abc', 'mno', 'xyz'];
    const result = getTopResults(list, 10);
    expect(result).toHaveLength(3);
    expect(result).toMatchSnapshot();
  });

  test('WHEN list EMPTY than return empty array', () => {
    const list: any[] = [];
    const result = getTopResults(list, 5);
    expect(result).toHaveLength(0);
    expect(result).toMatchSnapshot();
  });
});
