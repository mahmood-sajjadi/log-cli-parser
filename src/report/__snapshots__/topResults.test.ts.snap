// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTopResults WHEN list BIGGER than request PROVIDE what requested 1`] = `
[
  "abc",
  "mno",
]
`;

exports[`getTopResults WHEN list EMPTY than return empty array 1`] = `[]`;

exports[`getTopResults WHEN list SMALLER than request PROVIDE entire array 1`] = `
[
  "abc",
  "mno",
  "xyz",
]
`;
